pkg/controller/job/job_controller.go

```diff
@@ -1605,6 +1605,7 @@ func (jm *Controller) manageJob(ctx context.Context, job *batch.Job, jobCtx *syn
 						atomic.AddInt32(&active, -1)
 						errCh <- err
 					}
+					recordJobPodsCreationTotal(job)
```

Comment:
Updating metrics can be expensive. I would suggest you only increase them once per batch.

Either failures shouldn't count, or we should have a label indicating what was the status (failed/succeeded). I prefer the later.

In the current implementation, it is possible that a Pod creation counts more than once, if the first attempt fails.